query-coordinator/src/main/scala/com/socrata/querycoordinator/rollups/ClauseAwareExpressionRewriter.scala

```diff
@@ -0,0 +1,24 @@
+package com.socrata.querycoordinator.rollups
+
+import com.socrata.querycoordinator.rollups.QueryRewriter.{Analysis, Expr}
+import com.socrata.querycoordinator.rollups.QueryRewriterImplementation.{FunctionCall, Where}
+import com.socrata.querycoordinator.util.Join.NoQualifier
+import com.socrata.soql.typed
+
+class ClauseAwareExpressionRewriter(override val rollupColumnId: (Int) => String,
+                                    override val rewriteWhere: (Option[Expr], Analysis, Map[Expr, Int]) => Option[Where]) extends ExpressionRewriter(rollupColumnId, rewriteWhere) {
+  /** Recursively maps the Expr based on the rollupColIdx map, returning either
```

Comment:
I think it would be good to modify this comment to explain the specific case we are handling here (i.e. this class can be used in the so and so scenario).